contrib/masterfiles-stage/common.sh

```diff
@@ -12,146 +12,217 @@ if [ ! "$debug_mode" == "true" ]; then
   exec 2>&1
 fi
 
-function error_exit {
+error_exit() {
     # Display error message and exit
     echo "${0}: ${1:-"Unknown Error"}" 1>&2
     exit 1
 }
 
-git_branch_masterstage() {
-  # This function is designed to stage masterfiles from a git BRANCH
-  #   - Ensure git checkout exists, if it does not remove
-  #     obstructions and create new clone
-  #   - Ensure the origin is set properly
-  #   - Fetch updates
-  #   - Stash any local changes
-  #   - Checkout the proper branch
-  #   - Hard reset to remove any changes
-  #   - Validate the staged policy
-  #   - If Staged policy has validated and the release IDs are different
-  #     synchronize to final location (should we sync anyway to make sure the
-  #     distribution point is good?)
-
-  if ! type "git" > /dev/null; then
-    error_exit "git not found on path: ${PATH}"
-  fi
+set_staging_dir_from_params() {
+  # We probably want a different temporary location for each remote repository
+  # so that we can avoid conflicts and potential confusion.
+  # Example:
+  # ROOT="/opt/cfengine/masterfiles_staging"
+  # PARAMS="/var/cfengine/policychannel/production_1.sh"
+  # STAGING_DIR=/opt/cfengine/masterfiles/staging/_tmp_var_cfengine_policychannel_production_1_sh
 
-  # If we have a git checkout ensure the origin is set properly, update it and
-  # make sure all modified, extra, or missing files are reset so that we have a
-  # clean checkout
-
-  if [ -d "${STAGING_DIR}/.git" ]; then
-      cd "${STAGING_DIR}" && git remote set-url origin "${GIT_URL}" && (
-          git fetch -q origin || error_exit "Failed: git fetch -q origin"
-          git stash -q  || error_exit "Failed: git stash -q"
-          git checkout -q "${GIT_BRANCH}" || error_exit "Failed: git checkout -q ${GIT_BRANCH}"
-          git reset -q --hard "origin/${GIT_BRANCH}" || error_exit "Failed: git reset -q --hard origin/${GIT_BRANCH}"
-      ) || error_exit "Failed staging git branch"
-  else
-      if [ "$debug_mode" == "true" ]; then
-        echo "No git repo found in '${STAGING_DIR}'. Purging directory contents to clear path for fresh clone."
-      fi
-      rm -rf "${STAGING_DIR}"/* "${STAGING_DIR}"/.??*
-      if [ "$debug_mode" == "true" ]; then
-        echo "Cloning git repository '${GIT_URL}' '${GIT_BRANCH}' branch into '${STAGING_DIR}'"
-      fi
-      git clone --no-hardlinks "${GIT_URL}" "${STAGING_DIR}"
-
-      if [ "$debug_mode" == "true" ]; then
-        echo "Checking out '${GIT_BRANCH}' branch."
-      fi
-      cd "${STAGING_DIR}" && git checkout "${GIT_BRANCH}"
-  fi
+  STAGING_DIR="${ROOT}/_tmp$(echo "$PARAMS" | tr [./] _)"
+}
 
-  if /var/cfengine/bin/cf-promises -T "${STAGING_DIR}"; then
-    if ! /var/cfengine/bin/cf-promises -cf "${STAGING_DIR}/update.cf"; then
-        error_exit "Update policy staged in ${STAGING_DIR} could not be validated, aborting."
-    fi
+check_git_installed() {
+  git --version >/dev/null 2>&1 || error_exit "git not found on path: '${PATH}'"
```

Comment:
I did my testing with git version 1.7.1, which is roughly 6 years old and considered an "old version" according to [Wikipedia](https://en.wikipedia.org/wiki/Git_(software)#History).  So I think we're okay.